src/openvic-simulation/GameManager.cpp

```diff
@@ -105,21 +104,30 @@ static constexpr colour_argb_t DEFAULT_COLOUR_WHITE = (0xFFFFFF_argb).with_alpha
  * national focus, RGO, population density, sphere of influence, ranking and migration. */
 static constexpr colour_argb_t DEFAULT_COLOUR_GREY = (0x7F7F7F_argb).with_alpha(ALPHA_VALUE);
 
-template<IsColour ColourT = colour_t, std::derived_from<_HasColour<ColourT>> T>
-static constexpr auto get_colour_mapmode(T const*(Province::*get_item)() const) {
-	return [get_item](Map const& map, Province const& province) -> Mapmode::base_stripe_t {
-		T const* item = (province.*get_item)();
+template<utility::is_derived_from_specialization_of<_HasColour> T, typename P>
+requires(std::same_as<P, ProvinceDefinition> || std::same_as<P, ProvinceInstance>)
+static constexpr auto get_colour_mapmode(T const*(P::*get_item)() const) {
+	return [get_item](Map const& map, ProvinceInstance const& province) -> Mapmode::base_stripe_t {
+		ProvinceDefinition const& province_definition = province.get_province_definition();
+
+		T const* item;
+		if constexpr (std::same_as<P, ProvinceDefinition>) {
+			item = (province_definition.*get_item)();
+		} else {
+			item = (province.*get_item)();
+		}
```

Comment:
```suggestion
		T const& item = [&]{
		  if constexpr (std::same_as<P, ProvinceDefinition>) {
			  return *(province_definition.*get_item)();
		  } else {
			  return *(province.*get_item)();
		  }
		}();
```
Unless the item pointer needs to be changed, initializing them this way is generally safer.